cassiopeia/simulator/CellSubsampler.py

```diff
@@ -0,0 +1,124 @@
+import abc
+import networkx as nx
+import numpy as np
+from typing import Optional
+
+from cassiopeia.data import CassiopeiaTree
+
+
+class CellSubsamplerError(Exception):
+    """An Exception class for the CellSubsampler class."""
+
+    pass
+
+
+class CellSubsampler(abc.ABC):
+    r"""
+    Abstract base class for all cell samplers.
+
+    A CellSubsampler implements a method 'subsample' which, given a Tree,
```

Comment:
`Tree` -> `CassiopeiaTree`.

What information does this subsampler preserve? If this is done _before_ a tree has been populated, do we just subsample the character matrix?

If the tree has been populated, do we modify both the character matrix and the tree?